### PR TITLE
Install better_errors in development for awesome debugging

### PR DESCRIPTION
**Pull request by henare**
[better_errors](https://github.com/charliesome/better_errors#readme) is pretty fantastic. It would be handy to have this in development.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -88,4 +88,6 @@ group :development do
   gem 'mailcatcher', '~> 0.5.12'
   gem 'quiet_assets', '~> 1.1.0'
   gem 'rdoc', '~> 3.12.2'
+  gem 'better_errors'
```
Comment (garethrees) [inline Gemfile]: - Should add a [pessimistic version constraint](http://guides.rubygems.org/patterns/#pessimistic-version-constraint) to all gems
- better_errors has [dropped 1.9 support](https://github.com/charliesome/better_errors/pull/237) – should add a `:platforms => [:ruby_20]` constraint. Unfortunately `:ruby_20` is the latest version of ruby [our version of bundler](https://github.com/bundler/bundler/blob/master/CHANGELOG.md#130pre3-dec-21-2012) supports.

File: Gemfile [hunk 2]
```diff
@@ -88,4 +88,6 @@ group :development do
...
+  gem 'binding_of_caller'
```
Comment (garethrees) [inline Gemfile]: Just needs a pessimistic version constraint

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -149,8 +149,8 @@ def clear_session_credentials
 
   def render_exception(exception)
     # In development or the admin interface let Rails handle the exception
-    # with its stack trace templates
-    if Rails.application.config.consider_all_requests_local || show_rails_exceptions?
+    # unless it's for the favicon as this breaks the REPL in better_errors
+    if (Rails.application.config.consider_all_requests_local || show_rails_exceptions?) && request.path != "/favicon.ico"
```
Comment (garethrees) [inline app/controllers/application_controller.rb]: Could we just add the favicon to the admin layout?

Comment (garethrees) [inline app/controllers/application_controller.rb]: Added in https://github.com/mysociety/alaveteli/pull/3492

File: Gemfile (hunk 1 above)
Comment (henare) [inline Gemfile]: >Should add a pessimistic version constraint to all gems

Can this be added to the contribution guide (#2660)? I've never seen this done to every gem in a project before and I'd be genuinely interested in the reasoning.

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (henare) [inline app/controllers/application_controller.rb]: >Could we just add the favicon to the admin layout?

Nope :frowning: It seems the browser still tries to fetch `/favicon.ico`, which results in Alaveteli throwing an exception and thereby breaking the better_errors REPL.

**Comment by lizconlan** [comment]
closed in favour of #3508

**Pull request closed without merging**